src/sql/azdata.d.ts

```diff
@@ -67,6 +73,128 @@ declare module 'azdata' {
 		export const onDidChangeLanguageFlavor: vscode.Event<DidChangeLanguageFlavorParams>;
 	}
 
+	/**
+	 * The SQL Server data types.
+	 */
+	export type SqlDbType = 'BigInt' | 'Binary' | 'Bit' | 'Char' | 'DateTime' | 'Decimal'
+		| 'Float' | 'Image' | 'Int' | 'Money' | 'NChar' | 'NText' | 'NVarChar' | 'Real'
+		| 'UniqueIdentifier' | 'SmallDateTime' | 'SmallInt' | 'SmallMoney' | 'Text' | 'Timestamp'
+		| 'TinyInt' | 'VarBinary' | 'VarChar' | 'Variant' | 'Xml' | 'Udt' | 'Structured' | 'Date'
+		| 'Time' | 'DateTime2' | 'DateTimeOffset';
+
+
+	/**
+	 * The column information of a data set.
+	 */
+	export interface SimpleColumnInfo {
+		/**
+		 * The column name.
+		 */
+		name: string;
+		/**
+		 * This is expected to match the SqlDbTypes for serialization purposes.
+		 */
+		dataTypeName: SqlDbType;
```

Comment:
So this means it's tied specifically to MS SQL - is that what we want? Maybe we should first look into updating this to be more generic (such as `SimpleColumnInfo<T>` or have dataTypeName be a simple string instead.